lib/propolis/src/tasks.rs

```diff
@@ -436,20 +436,21 @@ impl TaskGroup {
 
     /// Block until all held tasks have been joined, returning any resulting
     /// [task::JoinError]s after doing so.
-    pub fn block_until_joined(&self) -> Option<Vec<task::JoinError>> {
-        let mut guard = self.0.lock().unwrap();
-        let workers = std::mem::replace(&mut *guard, Vec::new());
+    pub async fn block_until_joined(&self) -> Option<Vec<task::JoinError>> {
```

Comment:
nit: since this is an `async fn` now, it probably should no longer be called `block_until_joined`, since it, you know...doesn't block. maybe `wait_until_joined`? (and perhaps update the doc comment as well)

Reply:
Fixed in c4c80f2.